### ISSUE TITLE
ProjectModelTasks APIs fail in Nuke.Common version 6.0

### ISSUE THREAD

**Issue by nick-beer** [nuke-build/nuke#844]
### Usage Information

Nuke: 6.0 / SDK: 6.0 / TargetFramework: 6.0 / Windows 11

### Description

I'm in the process of updating my references to Nuke.Common in hopes to resolve #812 .  However, after updating to v 6.0.0 of Nuke (or 6.0.1), all the `ProjectModelTasks` APIs fail with:
```
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Build, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.                                                                                                                                                                                                           File name: 'Microsoft.Build, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' 
```

I can add an explicit PackageReference to that library in my project, but I then get the following failure:
```
Unhandled exception. Microsoft.Build.Exceptions.InvalidProjectFileException: The SDK 'Microsoft.NET.Sdk' specified could not be found.
```

### Reproduction Steps

1. New console app from template (.NET 6 SDK)
2. Add `<PackageReference Include="Nuke.Common" Version="6.0.1" />`
3. Change Program.cs to `var p = Nuke.Common.ProjectModel.ProjectModelTasks.ParseProject("some_csproj.csproj");`
4. Run

### Expected Behavior

`var p` from the reproduction steps would result in a valid p - no exceptions.

### Actual Behavior

```
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Build, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.                                                                                                                                                                                                           File name: 'Microsoft.Build, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' 
```

### Regression?

Yes - introduced in 6.0

### Known Workarounds

unknown

**Comment by matkoch** [comment]
Works for me and on CI: https://github.com/matkoch/NukeIssue844/runs/4820733295?check_suite_focus=true

**Comment by nick-beer** [comment]
I just tried a new Windows Sandbox instance, installed the .NET 6 SDK, then followed the steps to reproduce listed above and got the same error (the listed FileNotFoundException).

Additionally, in [this change](https://github.com/nuke-build/nuke/commit/65fcd49c7cba9901c4ad1f4b30225de966ed0187) the `runtime` assets were excluded from the `Microsoft.Build` PackageReference - so it's not clear to me where that Assembly should be coming from if not the NuGet package?

**Comment by matkoch** [comment]
they should come from the sdk installation

**Comment by nick-beer** [comment]
Thanks for being so patient with me - I figured out why I'm seeing the issue and you're not.  In both my simple repro case (and in the more complicated actual use case), I was attempting to use `ProjectModelTasks` prior to any other significant Nuke code running.  So, `ProjectModelTasks.Initialize()` never got the chance to run and find the `Microsoft.Build` library.  The `ProjectModelTasks` static constructor doesn't get a chance to run, presumably because the return type (Project) must be resolved first?

So, the workaround/fix for me is simple - just call `ProjectModelTasks.Initialize()` prior to using the API.

As a bonus, I learned about `MSBuildLocator` today - a pretty cool library.

Thanks again!
